### ISSUE TITLE
RethinkDB driver breaks Node.js domain

### ISSUE THREAD

**Issue by svarogg** [rethinkdb/rethinkdb#4475]
It looks like calling .run breaks the domain in which Node.js is, invalidating all data that was stored in it.

A quick-fix looks like wrapping the callback in ast.js/run with the following function:
https://gist.github.com/svarogg/1552cd4e9dd675494021

I don't know coffeescript, and don't have the time to fully study the consequences, so am not submitting a pull request at this time.

Thanks for your assistance.

**Comment by mlucy** [comment]
Ping @deontologician.

**Comment by deontologician** [comment]
> calling .run breaks the domain in which Node.js is, invalidating all data that was stored in it.

@svarogg could you give an example showing what you're running into?

**Comment by svarogg** [comment]
@deontologician I have initially encountered this problem when using the loopback rethinkdb connector. I have opened a ticket in the connector's repository: https://github.com/fuwaneko/loopback-connector-rethinkdb/issues/7 but after some investigation I have seen that the fix could be in the driver (all details in the link).

Nevertheless, when writing a shorter example to demonstrate this issue, I have seen that without the connector things work:
https://gist.github.com/perfectedTech/12e8156e9e882057ca0a

I must admit that I no longer fully understand the problem, and nodejs domains are new to me, but the the conjunction of loopback and rethinkdb definitely creates a problem.

Please advise. If you believe this is not a driver problem, this can be fixed inside the connector.
Thanks.

**Comment by deontologician** [comment]
I'm not familiar with node domains myself, but I'm a bit wary of modifying the code if there isn't an identifiable error. I can look into it more deeply later and make a more informed decision on it. For now I'm going to put it in the backlog, but if you are able to reproduce the error with the driver let me know and we can bump up the priority.

Thanks for reporting @svarogg

**Comment by svarogg** [comment]
@deontologician Hey there.
After some more research I was able to demonstrate this problem with a simple script, with only rethinkdb as a dependency.
https://gist.github.com/svarogg/7d3b0ac47f1833e93a27

**Comment by neumino** [comment]
This is actually an issue "in" bluebird
https://github.com/petkaantonov/bluebird/pull/664

**Comment by svarogg** [comment]
Looks like this was fixed in bluebird.

**Comment by deontologician** [comment]
OK, so it looks like we should update our bluebird dependency to > 2.9.31 to get these benefits.

**Comment by larkost** [comment]
@danielmewes: should we try and get this in for 2.0.4? 2.1? Both? I don't imagine that we will have many problems with this since there were none for our May 4 bump (#4178), and that was a bigger bump.

**Comment by danielmewes** [comment]
2.1 is enough I think.

**Comment by danielmewes** [comment]
@AtnNn what steps to we need to take to bump the bluebird dependency?

**Comment by AtnNn** [comment]
```
$ vim mk/support/pkg/bluebird.sh
$ git diff
-version=2.9.25
+version=2.9.32
$ ./configure
$ make shrinkwrap-bluebird
$ git diff --stat
 mk/support/pkg/bluebird.sh         | 2 +-
 mk/support/pkg/bluebird.shrinkwrap | 8 ++++----
```

**Comment by danielmewes** [comment]
The bluebird version change is in code review 3058.

**Comment by danielmewes** [comment]
Never mind, it already was in review 3055

**Comment by AtnNn** [comment]
Sorry, I should have mentioned that I had put it in review.

Fixed in next (3695c3b).
